pkg/glob/glob.go

```diff
@@ -63,11 +68,14 @@ func glob(segs []Segment, dir string, cb func(string) bool) bool {
 	}
 
 	if len(segs) == 0 {
-		return cb(dir)
+		if info, err := os.Stat(dir); err == nil {
```

Comment:
You can save the `FileInfo` the code on line 65 obtained earlier and avoid another `Stat`.

Reply:
I have no idea what you're suggesting. If `len(segs) == 0` was true when this function was entered then the preceding `os.Stat()` would not have been run.

Reply:
`len(segs) == 0` will only become true if the loop has been run, which shortens `segs` (`segs = segs[2:]`), and this function is never called with an empty `segs`.

OTOH, it's not easy to reason all that. The code as it is now is easier to reason about.